numpyro/distributions/discrete.py

```diff
@@ -301,6 +303,43 @@ def Categorical(probs=None, logits=None, validate_args=None):
         raise ValueError('One of `probs` or `logits` must be specified.')
 
 
+@copy_docs_from(Distribution)
+class Delta(Distribution):
+    arg_constraints = {'value': constraints.real, 'log_density': constraints.real}
+    support = constraints.real
+
+    def __init__(self, value=0., log_density=0., event_dim=0, validate_args=None):
```

Comment:
nit: the argument name `event_dim` is misleading, since this refers to the number of event dims from the right and not a single dimension. A better name might be `event_ndims` or `event_dims`.

Reply:
Yup, at first I used `event_ndim` but later decide that just keep the same interface as in Pyro. Will change back to ndim.